src/main/java/edu/ie3/datamodel/io/sink/CsvFileSink.java

```diff
@@ -324,18 +324,19 @@ private <C extends UniqueEntity> void write(C entity) {
     LinkedHashMap<String, String> entityFieldData;
     try {
       entityFieldData =
-          processorProvider
-              .handleEntity(entity)
-              .orElseThrow(
-                  () ->
-                      new SinkException(
-                          "Cannot persist entity of type '"
-                              + entity.getClass().getSimpleName()
-                              + "'. This sink can only process the following entities: ["
-                              + processorProvider.getRegisteredClasses().stream()
-                                  .map(Class::getSimpleName)
-                                  .collect(Collectors.joining(","))
-                              + "]"));
+          csvEntityFieldData(
+              processorProvider
+                  .handleEntity(entity)
```

Comment:
I'd rather apply `csvEntityFieldData` via `map` here so that indentation does not get too crazy. So it'd be
```
.handleEntity(entity)
.map(csvEntityFieldData)
.orElseThrow(...
```
or similar.

Reply:
Fair point !